packages/backend/src/api/user/userRouter.ts

```diff
@@ -34,11 +41,93 @@ export const userRouter: Router = (() => {
     responses: createApiResponse(UserSchema, 'Success'),
   });
 
-  router.get('/:id', validateRequest(GetUserSchema), async (req: Request, res: Response) => {
-    const id = parseInt(req.params.id as string, 10);
+  router.get('/:id', async (req: Request, res: Response) => {
+    const id = req.params.id;
     const serviceResponse = await userService.findById(id);
     handleServiceResponse(serviceResponse, res);
   });
 
+  router.post('/', async (req: Request, res: Response) => {
+    const { email, username, password, full_name, phone_number, level_id, category_id } = req.body;
+    const hashPassword = await bcrypt.hash(password, 1);
+    const serviceResponse = await userService.createUser(
+      email,
+      username,
+      hashPassword,
+      full_name,
+      phone_number,
+      level_id,
+      category_id
+    );
+    handleServiceResponse(serviceResponse, res);
+  });
+
+  router.delete('/:id', authenticate, async (req: Request, res: Response) => {
+    const id = req.params.id;
+    if (req.user && req.user.id === id) {
+      //making sure that the authenticated user is deleting his own account and not someone else's account
+      const serviceResponse = await userService.deleteUser(id);
+      handleServiceResponse(serviceResponse, res);
+    } else {
+      res.status(401).json({ message: 'Unauthorized' }); //do like this or create a serviceResponse object, assign it the error code and message and send it to handleserviceresponse?
+    }
+  });
+  router.put('/:id', authenticate, async (req: Request, res: Response) => {
+    const { full_name, level_id, category_id, tags } = req.body;
+    const id = req.params.id;
+    if (req.user && req.user.id === id) {
+      const serviceResponse = await userService.updateUser(id, full_name, level_id, category_id, tags);
+      handleServiceResponse(serviceResponse, res);
+    } else {
+      res.status(401).json({ message: 'Unauthorized' }); //do like this or create a serviceResponse object, assign it the error code and message and send it to handleserviceresponse?
+    }
+  });
+
+  router.post('/login', async (req, res, next) => {
+    passport.authenticate('local', { session: false }, (err, user, info) => {
```

Comment:
We need to move the authenticate logic to a different service. The routes/controllers should not be responsible for auth logic.